src/request.rs

```diff
@@ -1,46 +1,102 @@
 //! creating and sending HTTP requests
 use crate::{
     error,
-    response::{Headers, Response, CR_LF_2},
+    response::{find_slice, Headers, Response, CR_LF_2},
     tls,
     uri::Uri,
 };
 use std::{
     fmt,
-    io::{self, Read, Write},
+    io::{self, ErrorKind, Read, Write},
     net::{TcpStream, ToSocketAddrs},
     path::Path,
-    time::Duration,
+    time::{Duration, Instant},
 };
 
 const CR_LF: &str = "\r\n";
+const BUF_SIZE: usize = 8 * 1024;
+const SMALL_BUF_SIZE: usize = 8 * 10;
+const TEST_FREQ: usize = 200;
+const CONNECTION_TIMEOUT: u64 = 3 * 3600;
 
-///Copies data from `reader` to `writer` until the specified `val`ue is reached.
+///Copies data from `reader` to `writer` until the `timeout` is reached.
 ///Returns how many bytes has been read.
-pub fn copy_until<R, W>(reader: &mut R, writer: &mut W, val: &[u8]) -> Result<usize, io::Error>
+pub fn copy_with_timeout<R, W>(reader: &mut R, writer: &mut W, timeout: Instant) -> io::Result<u64>
```

Comment:
`timeout` here should be called `deadline`. Timeout would be of type `Duration`, while this is an `Instant` by which you need it to be done.

Reply:
You're right. I've renamed it